crates/containerd-shim-wasm/src/test/signals.rs

```diff
@@ -0,0 +1,143 @@
+use std::future::pending;
+use std::sync::mpsc::channel;
+use std::sync::{Arc, LazyLock};
+use std::thread::sleep;
+use std::time::Duration;
+
+use anyhow::{bail, Result};
+use containerd_shim_wasm_test_modules::HELLO_WORLD;
+use tokio::sync::Notify;
+
+use crate::container::{Engine, Instance, RuntimeContext};
+use crate::sandbox::Stdio;
+use crate::testing::WasiTest;
+
+#[derive(Clone)]
+pub struct SomeEngine;
+
+impl Default for SomeEngine {
+    fn default() -> Self {
+        SomeEngine
+    }
+}
+
+async fn ctrl_c(use_libc: bool) {
+    static CANCELLATION: LazyLock<Notify> = LazyLock::new(|| Notify::new());
+
+    fn on_ctr_c(_: libc::c_int) {
+        CANCELLATION.notify_waiters();
+    }
+
+    if use_libc {
+        unsafe { libc::signal(libc::SIGINT, on_ctr_c as _) };
+        CANCELLATION.notified().await;
+    } else {
+        let _ = tokio::signal::ctrl_c().await;
+    }
+}
+
+impl Engine for SomeEngine {
+    fn name() -> &'static str {
+        "some-engine"
+    }
+
+    fn run_wasi(&self, ctx: &impl RuntimeContext, stdio: Stdio) -> Result<i32> {
+        stdio.redirect()?;
+        let name = ctx.entrypoint().func;
+        tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()?
+            .block_on(async move {
+                use tokio::time::sleep;
+                let use_libc = std::env::var("USE_LIBC").unwrap_or_default();
```

Comment:
we need to document how to run this test since it introduces a new environment variable "USE_LIBC". Do you want to add some instructions to the /docs?

Reply:
I'll add a commend in the test file.
Ideally we would fix the problem, and then we can remove the `libc` implementation, and remove the `ignore` attribute from the test.